### PR TITLE
[FIX] Avoid canceling transactions after submission

### PR DESCRIPTION
**Pull request by tommasini**
**Description**
When we put the app in the background we were trying to cancel the transaction. The transaction still succeeded because the transaction status it's not cancelable but shows one notification that says "Oops Something went wrong".

**Proposed solution**
The proposed solution is only to cancel the transaction if the status it's not, submitted, cancellable, approved, failed and rejected when the app goes to the background.

**Code Impact**
The code impact is low, since we just change the code when the app state (foreground, background), changes.

**Screenshots/Videos**
Bug:
https://recordit.co/Pqa7Uq34hv

Solution:
https://recordit.co/EkE0jdyCMj

**Test Cases**
Case1:
 - Go to browser and go to https://metamask.github.io/test-dapp/
 - Create a token
 - Put the mobile app in the background state
 - Watch the result of the notification
Case 2:
 - Go to browser and go to https://metamask.github.io/test-dapp/
 - Create a token
 - Put the mobile app in the background state
 - Reopen the mobile app
 - Watch the result of the notification
Case 3:
 - Go to browser and go to https://metamask.github.io/test-dapp/
 - Create a token
 - Watch the result of the notification

**Checklist**

* [x] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

**Issue**
Progresses #https://github.com/MetaMask/metamask-mobile/issues/3927

### CODE REVIEW

File: app/components/Views/Approval/index.js [hunk 1]
```diff
@@ -125,8 +126,13 @@ class Approval extends PureComponent {
...
 		try {
 			if (appState !== 'active') {
-				const { transaction } = this.props;
-				transaction && transaction.id && Engine.context.TransactionController.cancelTransaction(transaction.id);
+				const { transaction, transactions } = this.props;
+				const currentTransaction = transactions.find((tx) => tx.id === transaction.id);
+
+				transaction &&
+					transaction.id &&
+					currentTransaction.status !== TX_SUBMITTED &&
```
Comment (gantunesr) [inline app/components/Views/Approval/index.js:134]: `Reject`, `Cancelled`, `Failed` and `Confirmed` are final states. You should exclude them too

**Review by gantunesr — CHANGES_REQUESTED** [review]

**Review by gantunesr — APPROVED** [review]
LGTM

**Comment by tommasini** [comment]
Test case 1:
 - Go to browser and go to https://metamask.github.io/test-dapp/
 - Create a token
 - Put the mobile app in the background state
 - Watch the result of the notification
Test case 2:
 - Go to browser and go to https://metamask.github.io/test-dapp/
 - Create a token
 - Put the mobile app in the background state
 - Reopen the mobile app
 - Watch the result of the notification
 Test case 3:
 - Go to browser and go to https://metamask.github.io/test-dapp/
 - Create a token
 - Watch the result of the notification

**Comment by Jens05** [comment]
Any progress on this? If this issue is resolved i can continue with my Dapp :)

**Review by cortisiko — APPROVED** [review]
This ticket is QA passed. Scenarios I tested:

Send transaction 
Swap transaction
dapps transactions. 

I no longer see the `ooops something went wrong` alert message when the app was in the background. I also did not see that alert message when I killed and relaunched the app. Finally with push notifications turned on I did not receive a `ooops something went wrong` notification. Good work @tommasini 🌮 🌮

**Pull request merged**